core/jvm/src/main/scala/fs2/compress.scala

```diff
@@ -25,45 +22,74 @@ object compress {
       nowrap: Boolean = false,
       bufferSize: Int = 1024 * 32,
       strategy: Int = Deflater.DEFAULT_STRATEGY
-  ): Pipe[F, Byte, Byte] = { in =>
-    Pull.suspend {
-      val deflater = new Deflater(level, nowrap)
-      deflater.setStrategy(strategy)
-      val buffer = new Array[Byte](bufferSize)
-      _deflate_stream(deflater, buffer)(in)
-    }.stream
+  ): Pipe[F, Byte, Byte] = stream => {
+    val deflater = new Deflater(level, nowrap)
```

Comment:
Looks like this needs a `Stream.suspend` to ensure each use gets its own `deflater` instance. E.g., consider `val p = compress.deflate(); val x = s.through(p); val y = x.toList, val z = x.toList`

Reply:
I'm not sure. Deflate and inflate return a function so a new Deflater is created for each use. I've added tests "deflate and inflate are reusable" and "gzip and gunzip are reusable." Let me know if I've misunderstood. Thanks

Reply:
In the example, the function is only called once `val x = s.through(p)` and then "run" twice. Here's a simplified example:

```scala
@ def pipe[F[_]]: Pipe[F, Int, Int] = in => {
    var cnt = 0
    in.map { i => cnt += 1; i } ++ Stream(cnt)
  }
defined function pipe

@ Stream(1, 1, 1).through(pipe).toList
res2: List[Int] = List(1, 1, 1, 3)

@ val x = Stream(1, 1, 1).through(pipe)
x: Stream[Nothing, Int] = Stream(..)

@ x.toList
res4: List[Int] = List(1, 1, 1, 3)

@ x.toList
res5: List[Int] = List(1, 1, 1, 6)

@ x.toList
res6: List[Int] = List(1, 1, 1, 9)
```

Reply:
Ah, thanks for the second explanation. I expert this will be addressed by your suggestion to use Sync.delay, but I'll make sure this case is explicitly confirmed by the tests.